.github/workflows/upgrade.yml

```diff
@@ -0,0 +1,34 @@
+name: Upgrade Niv and LTS
+
+on:
+  schedule:
+    # Second and fourth Tuesday of every month at 8am
+    - cron: '00 08 8-14,22-28 * 2'
+  workflow_dispatch:
+  push:
+    branches:
+      - upgrade-bot
+jobs:
+  create-pull-request:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - uses: cachix/install-nix-action@v12
+    - uses: cachix/cachix-action@v7
+      with:
+        name: earnestresearch-public
+    - name: Update dependencies
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      run: GITHUB_PATH= nix-shell --command 'niv update'
+    - name: Update test cabal.project.freeze to latest LTS
+      run: curl -L https://www.stackage.org/lts/cabal.config > test/cabal.project.freeze
+    - name: Create pull request
```

Comment:
We need to give this a personal access token.  It defaults to `GITHUB_TOKEN`, which doesn't have the power to trigger an action that triggers other actions.  That means we get PRs like #72, but we don't get them checked.